src/site/content/en/blog/optimize-long-tasks/index.md

```diff
@@ -0,0 +1,403 @@
+
+`isInputPending` is a function you can run at any time to determine if the user is attempting to interact with a page element: a call to `isInputPending` will return `true`. It returns `false` otherwise.
+
+Say you have a queue of tasks you need to run, but you don't want to get in the way of any inputs, such as an attempt to open a navigation menu, for example. This code which uses both `isInputPending` and our custom `yieldToMain()` function ensures that the input won't be delayed while the user is trying to interact with the page:
+
+```js
+async function saveSettings () {
+  // A task queue of functions
+  const tasks = [
+    validateForm,
+    showSpinner,
+    saveToDatabase,
+    updateUI,
+    sendAnalytics
+  ];
+  
+  while (tasks.length > 0) {
+    // Yield to a pending user input:
+    if (navigator.scheduling.isInputPending()) {
+      // There's a pending user input. Yield here:
+      await yieldToMain();
+    } else {
+      // Shift the the task out of the queue:
+      const task = tasks.shift();
+
+      // Run the task:
+      task();
+    }
+  }
+}
+```
+
+While `saveSettings` runs, it will loop over the tasks in the queue. If `isInputPending` ever returns `true` during the loop, `saveSettings` will call `yield` so the user input can be handled. Otherwise, it will shift the next task off the front of the queue and run it continuously. It will do this until no more tasks are left.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/snMl3kRlWyJjdbL0qsqM.png", alt="A depiction of the saveSettings function running in Chrome's performance profiler. The resulting task blocks the main thread until isInputPending returns true, in which case, the task yields to the main thread.", width="800", height="254" %}
+  <figcaption>
+    <code>saveSettings</code> runs a task queue for five tasks, but the user has clicked to open a menu while the second work item was running. <code>isInputPending</code> yields to the main thread to handle the interaction, and resume running the rest of the tasks.
+  </figcaption>
+</figure>
+
+{% Aside %}
+`isInputPending()` may not always return `true` immediately after user input. This is because it takes time for the operating system to tell the browser that the interaction occurred. This means that other code may have already started executing (as you can see with the `saveToDatabase` function in the above screenshot). Even if you use `isInputPending()` it's still crucial that you limit the amount of work you do in each function.
+{% endAside %}
+
+`isInputPending` in combination with a yielding mechanism is a great way to get the browser to stop whatever tasks it's processing on so that it can respond to critical user-facing interactions. That can help improve your page's ability to respond to the user in many situations when a lot of tasks are in flight.
```

Comment:
```suggestion
`isInputPending()` in combination with a yielding mechanism is a great way to get the browser to stop whatever tasks it's processing on so that it can respond to critical user-facing interactions. That can help improve your page's ability to respond to the user in many situations when a lot of tasks are in flight.
```